components/sidebar_right/sidebar_right.tsx

```diff
@@ -1,49 +1,58 @@
+import {Channel} from 'mattermost-redux/types/channels';
+
+import {trackEvent} from 'actions/telemetry_actions';
 import Constants from 'utils/constants';
-import * as Utils from 'utils/utils.jsx';
+import * as Utils from 'utils/utils';
 
 import FileUploadOverlay from 'components/file_upload_overlay';
 import RhsThread from 'components/rhs_thread';
 import RhsCard from 'components/rhs_card';
-import Search from 'components/search/index.tsx';
+import Search from 'components/search/index';
 
 import RhsPlugin from 'plugins/rhs_plugin';
 
-export default class SidebarRight extends React.PureComponent {
-    static propTypes = {
-        isExpanded: PropTypes.bool.isRequired,
-        isOpen: PropTypes.bool.isRequired,
-        currentUserId: PropTypes.string.isRequired,
-        channel: PropTypes.object,
-        postRightVisible: PropTypes.bool,
-        postCardVisible: PropTypes.bool,
-        searchVisible: PropTypes.bool,
-        isMentionSearch: PropTypes.bool,
-        isFlaggedPosts: PropTypes.bool,
-        isPinnedPosts: PropTypes.bool,
-        isPluginView: PropTypes.bool,
-        previousRhsState: PropTypes.string,
-        rhsChannel: PropTypes.object,
-        selectedPostId: PropTypes.string,
-        selectedPostCardId: PropTypes.string,
-        actions: PropTypes.shape({
-            setRhsExpanded: PropTypes.func.isRequired,
-            showPinnedPosts: PropTypes.func.isRequired,
-            openRHSSearch: PropTypes.func.isRequired,
-            closeRightHandSide: PropTypes.func.isRequired,
-            openAtPrevious: PropTypes.func.isRequired,
-            updateSearchTerms: PropTypes.func.isRequired,
-        }),
+export type Props = {
+    isExpanded: boolean;
+    isOpen: boolean;
+    currentUserId: string;
+    channel?: Channel;
+    postRightVisible?: boolean;
+    postCardVisible?: boolean;
+    searchVisible: boolean;
+    isMentionSearch?: boolean;
+    isFlaggedPosts?: boolean;
+    isPinnedPosts?: boolean;
+    isPluginView?: boolean;
+    previousRhsState?: string;
+    rhsChannel: Channel;
+    selectedPostId?: string;
+    selectedPostCardId?: string;
+    actions: {
+        setRhsExpanded: (expanded: boolean) => void;
+        showPinnedPosts: (channelId?: string) => void;
+        openRHSSearch: () => void;
+        closeRightHandSide: () => void;
+        openAtPrevious: (previous: unknown) => void;
+        updateSearchTerms: (terms: string) => void;
     };
+};
+
+export type State = {
+    isOpened: boolean;
+}
 
-    constructor(props) {
+export default class SidebarRight extends React.PureComponent<Props, State> {
+    sidebarRight: React.RefObject<HTMLDivElement>;
+    previous: { searchVisible: boolean | undefined; isMentionSearch: boolean | undefined; isPinnedPosts: boolean | undefined; isFlaggedPosts: boolean | undefined; selectedPostId: string | undefined; selectedPostCardId: string | undefined; previousRhsState: string | undefined } | undefined;
```

Comment:
You can use `fieldName?: type` instead of `fieldName: type | undefined` as a shorthand for optional parameters.

Reply:
It might be worth adding this type in the `types` folder since it's also used by the action that the `previous` field is called with.

Reply:
I have added in the types folder. Let me know if you need any changes or suggest. @devinbinnie